### ISSUE TITLE
Running out of memory when moving a large amount a rows from one table to another

### ISSUE THREAD

**Issue by Coriou** [joshcam/PHP-MySQLi-Database-Class#479]
I'm moving a large number of rows (each row has little content though, only INTs and smaller) from one table to another like so :

```
	/* Get data older than $days using $buffer as the limit to mitigate memory usage */
		while ( !$done ) {
			$this->db->where( 'time', $days, '<' );
			$q = $this->db->get( 'myTable', $buffer );

			if ( !$q ) {
				$done = true;
				break;
			}else{
			/* Moving that big ass amount of data */
				foreach ( $q as $c ) {

					$moved[] = $this->db->insert( 'myTable_old', $c );
					$this->db->where( 'id', $c['id'] );
					$del[] = $this->db->delete( 'myTable' );

				}
			}
		}
```

In my case the "buffer" is 1000 (which means I'm moving rows one thousand at a time) because if I try to do it all at once, I totally run out of memory (I have something like 9 million rows to move).

My problem is I eventually get a running out of memory fatal error and, pardon my lack of knowledge here, I can't figure out why. Here's the error :

`Fatal error: Allowed memory size of 536870912 bytes exhausted (tried to allocate 75 bytes) in /my/path.php on line 1172`

Line 1172 is within a loop, in the `_dynamicBindResults()` helper method :

```
              foreach ($row as $key => $val) {
                    if (is_array($val)) {
                        foreach ($val as $k => $v) {
                            $result[$key][$k] = $v;
                        }
                    } else {
                        $result[$key] = $val; // This is line 1172
                    }
                }
```

I don't know what's causing it to be honest, but I thought maybe it'd be worth reporting.

**Comment by avbdr** [comment]
have you tried to lower your buffer value to a lower, say 100-200?

**Comment by Coriou** [comment]
> have you tried to lower your buffer value to a lower, say 100-200?

Yes, I have tried that.

Actually, I believe it's a bug in my software now. Because what happens is the `$result` variable does not seem to be destroyed until my script has fully completed. In my opinion, it should be destroyed after every query, or after every batch of queries at least maybe.

What's causing the out of memory exception is that the array is eventually too big.

I haven't looked at back at this since I posted here. But I will eventually and update you if I find what's causing this array to never be destroyed.
